UI/MapWnd.cpp

```diff
@@ -1923,42 +1927,112 @@ void MapWnd::RenderSystems() {
+
             if (fog_scanlines
                 && (universe.GetObjectVisibilityByEmpire(system_icon.first, empire_id) <= VIS_BASIC_VISIBILITY))
             {
                 m_scanline_shader.SetColor(GetOptionsDB().Get<GG::Clr>("UI.system-fog-of-war-clr"));
                 m_scanline_shader.RenderCircle(circle_ul, circle_lr);
             }
 
-            // render circles around systems that have at least one starlane, if circles are enabled.
+            // render circles around systems that have at least one starlane, if they are enabled
             if (circles) {
                 if (std::shared_ptr<const System> system = GetSystem(system_icon.first)) {
                     if (system->NumStarlanes() > 0) {
-                        glColor(GetOptionsDB().Get<GG::Clr>("UI.unowned-starlane-colour"));
 
+                        bool has_empire_planet = false;
+                        bool has_neutrals = false;
+                        std::map<int, int> colony_count_by_empire_id;
+                        const std::set<int>& known_destroyed_object_ids = GetUniverse().EmpireKnownDestroyedObjectIDs(HumanClientApp::GetApp()->EmpireID());
+
+                        for (std::shared_ptr<const Planet> planet : Objects().FindObjects<const Planet>(system->PlanetIDs())) {
+                            if (known_destroyed_object_ids.count(planet->ID()) > 0)
+                                continue;
+
+                            // remember if this system has a player-owned planet, count # of colonies for each empire
+                            if (!planet->Unowned()) {
+                                has_empire_planet = true;
+
+                                std::map<int, int>::iterator it = colony_count_by_empire_id.find(planet->Owner()) ;
+                                if (it != colony_count_by_empire_id.end()) {
+                                    it->second++;
+                                } else {
+                                    colony_count_by_empire_id.insert({planet->Owner(), 1});
+                                }
+                            }
+
+                            // remember if this system has neutrals
+                            if (planet->Unowned() && !planet->SpeciesName().empty() && planet->CurrentMeterValue(METER_POPULATION) > 0.0) {
+                                has_neutrals = true;
+                            }
+                        }
+
+                        // draw outer circle in color of supplying empire
                         int empire_id = GetSupplyManager().EmpireThatCanSupplyAt(system_icon.first);
                         if (empire_id != ALL_EMPIRES) {
                             if (Empire* empire = GetEmpire(empire_id)) {
                                 glColor(empire->Color());
                             }
+                        } else {
+                            glColor(GetOptionsDB().Get<GG::Clr>("UI.unowned-starlane-colour"));
```

Comment:
It is, otherwise the circle will have a random previously used color (possibly set by the scanline renderer before). However above this, calling glColor was not necessary, so I removed that instead.